### ISSUE TITLE
Broadcast macro inside of @with doesn't seem to work anymore

### ISSUE THREAD

**Issue by oschulz** [joshday/PropertyUtils.jl#3]
Nowadays (PropertyUtils v1.3, Julia v1.9), this still works

```julia
julia> df = @inferred StructArray(
    a = rand(100), b = rand(100),
    c = rand(100), d = rand(100),
    e = rand(100), f = rand(100)
)

julia> @. @with df df[a * c * d < .25, :]
```

But this version, which should avoid unnecessary memory access (see #1) doesn't seem to work anymore:

```julia
julia> @with df @. df[a * c * d < .25, :]
ERROR: UndefVarError: `##337` not defined
Stacktrace:
 [1] top-level scope
   @ REPL[36]:1
```

**Comment by joshday** [comment]
I had some changes locally that I guess I never pushed/released.  I just pushed them and now your example errors with:

```Julia
julia> @with df @. df[a * c * d < .25, :]
ERROR: TypeError: non-boolean (BitVector) used in boolean context
```

---

I'm not sure this is fixable since I ditched `eval` in the implementation.

For each valid identifier `a`,  `@with` changes it to `hasproperty(df, :a) ? df.a : a`.  There's no way to hide that expression from `@.` so in turn it changes to `hasproperty.(df, :a) ? df.a : a`.  

Better docs may be the solution.
